components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/expmapper/InvalidRequestExceptionMapper.java

```diff
@@ -88,6 +92,41 @@ public Response toResponse(InvalidRequestParentException exception) {
                 }
                 return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
             }
+        } else if (exception instanceof TokenEndpointBadRequestException) {
+            try {
+                if (exception.getMessage() != null) {
+                    OAuthResponse oAuthResponse = OAuthASResponse.errorResponse(HttpServletResponse.SC_BAD_REQUEST)
```

Comment:
Shall we move building errorResponse to a private method so that we can reuse it?